lib/rules/no-implicit-coercion.js

```diff
@@ -224,7 +224,13 @@ module.exports = {
                 if (!operatorAllowed && options.boolean && isBinaryNegatingOfIndexOf(node)) {
                     const recommendation = `${sourceCode.getText(node.argument)} !== -1`;
 
-                    report(node, recommendation);
+                    context.report({
```

Comment:
This is a duplicate of lines 198 to 203. Instead of duplicating those lines, maybe just add an extra parameter to `report()` indicating if it should be fixed? (I just generally prefer to have messages centralized.)